app/controllers/api/braintree_controller.rb

```diff
@@ -1,31 +1,57 @@
 class Api::BraintreeController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
 
   def token
     render json: { token: ::Braintree::ClientToken.generate }
   end
 
   def transaction
-    sale = braintree.make_transaction(options)
+    result = braintree::Transaction.make_transaction(transaction_options)
+    if result.success?
+      render json: { success: true, transaction_id: result.transaction.id }
+    else
+      render json: { success: false, errors: result.errors }
```

Comment:
I've been using 422 in our controllers for validation failures. You can read some discussion [here](https://www.quora.com/Which-HTTP-code-is-best-suited-for-validation-errors-400-or-422), the gist is that 422 means "I can read and understand your value but it's not valid"